app/src/main/res/layout/custom_home_tile.xml

```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:orientation="vertical"
+    android:layout_width="140dp"
+    android:layout_height="100dp"
+    android:background="@drawable/home_tile_background"
+    android:foreground="@drawable/home_tile_selector"
+    android:focusable="true"
+    android:clickable="true"
+    android:clipChildren="false"
+    android:layout_marginStart="12dp"
+    android:layout_marginEnd="12dp"
+    android:layout_marginBottom="18dp">
+
+    <ImageView
+        android:id="@+id/home_tile_menu_icon"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="end|top"
+        android:layout_marginEnd="-10dp"
+        android:layout_marginTop="-10dp"
+        android:background="@drawable/context_menu_button_background"
+        android:elevation="4dp"
+        android:padding="4dp"
+        android:src="@drawable/ic_wide_menu_home"
+        android:visibility="gone" />
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_marginTop="4dp"
```

Comment:
I didn't realize these files would be so similar so I mislead you again – sorry about that. :(

I think in general having so much code duplication is bad practice. I think there are two approaches:
- Can you we have one layout? i.e. if we used one layout file with no dynamic changes, would Tif mind if the bundled tiles grew in size too?
- If she'd mind, your initial approach of changing the margins (and cornerRadius!) programmatically would have been better. You can `getLayoutMargins()`, update the values, and `setLayoutMargins()` (you may need to cast to MarginLayoutParams).

Do you mind trying these approaches?

fwiw, I have a bad habit of giving really general advice instead of advice specific to the problem someone is trying to solve (which happened here - XML is generally preferred over code) so I'll try to be better about this in the future.